airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/datetime_stream_slicer.py

```diff
@@ -188,14 +191,14 @@ def _parse_timedelta(cls, time_str):
         Parse a time string e.g. (2h13m) into a timedelta object.
         Modified from virhilo's answer at https://stackoverflow.com/a/4628148/851699
         :param time_str: A string identifying a duration. (eg. 2h13m)
-        :return datetime.timedelta: A datetime.timedelta object
+        :return relativedelta: A relativedelta object
         """
         parts = cls.timedelta_regex.match(time_str)
 
         assert parts is not None
 
         time_params = {name: float(param) for name, param in parts.groupdict().items() if param}
-        return datetime.timedelta(**time_params)
+        return relativedelta(**time_params)
```

Comment:
what's the motivation for using `relativedelta` instead of `datetime.timedelta`?

Reply:
`datetime.timedelta` supports from microseconds up to weeks, whereas `relativedelta` extends that up to years.